File: languages/rust/prime.rs
use std::io;

fn is_prime(n: i32) -> bool {
    if n <= 1 {
        return false;
    }
    if n == 2 {
        return true;
    }
    if n % 2 == 0 {
        return false;
    }
    
    let sqrt_n = (n as f64).sqrt() as i32;
    for i in (3..=sqrt_n).step_by(2) {
        if n % i == 0 {
            return false;
        }
    }
    true
}

fn main() {
    println!("Enter a number to check if it's prime:");

    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read input");
    
    let number: i32 = input.trim().parse().expect("Invalid input");

    if is_prime(number) {
        println!("{} is a prime number.", number);
    } else {
        println!("{} is not a prime number.", number);
    }
}

File: languages/julia/prime.jl
function is_prime(n::Int)
    if n <= 1
        return false
    elseif n == 2
        return true
    else
        for i in 2:floor(Int, sqrt(n))
            if n % i == 0
                return false
            end
        end
    end
    return true
end

println("Enter a number to check if it's prime: ")
n = parse(Int, readline())

if is_prime(n)
    println("$n is a prime number.")
else
    println("$n is not a prime number.")
end
